services/player-service/application/src/test/java/com/kjeldsen/player/application/usecases/GetHistoricalTrainingUseCaseTest.java

```diff
@@ -0,0 +1,45 @@
+package com.kjeldsen.player.application.usecases;
+
+import com.kjeldsen.player.domain.Player;
+import com.kjeldsen.player.domain.events.PlayerTrainingEvent;
+import com.kjeldsen.player.domain.repositories.PlayerReadRepository;
+import com.kjeldsen.player.domain.repositories.PlayerTrainingEventReadRepository;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class GetHistoricalTrainingUseCaseTest {
+
+    private final PlayerTrainingEventReadRepository mockPlayerTrainingEventReadRepository = Mockito.mock(PlayerTrainingEventReadRepository.class);
+    private final PlayerReadRepository mockPlayerReadRepository = Mockito.mock(PlayerReadRepository.class);
+    private final GetHistoricalTrainingUseCase getHistoricalTrainingUseCase = new GetHistoricalTrainingUseCase(mockPlayerTrainingEventReadRepository, mockPlayerReadRepository);
+
+    @Test
+    public void should_generate_runtimeException_if_player_not_found() {
+        Player.PlayerId playerId = new Player.PlayerId("fulano");
+        when(mockPlayerReadRepository.findOneById(playerId)).thenReturn(Optional.empty());
+        assertThrows(RuntimeException.class, () -> {
```

Comment:
specific exception message

Reply:
Ok, in this case i have to use at first the method "String.format" to indicate "PlayerId" to the follow message "Player not found with ID %s" because is the way that appear in the class.

Reply:
Ok, in this case i have to use at first the method "String.format" to indicate "PlayerId" to the follow message "Player not found with ID %s" because is the way that appear in the class.